### DISCUSSION TITLE
Why does https://maplibre.org/maputnik fetch http://localhost:8000/styles ?

### DISCUSSION THREAD

**Discussion by zstadler** [maputnik/editor#886] [Q&A]
To reproduce:
- Open the browser's development console 
- Load https://maplibre.org/maputnik 
- Look for localhost:8000 in the network tab

**Comment by HarelM** [comment]
Seems like this is related to the desktop communication.
See here:
https://github.com/maplibre/maputnik/blob/1da65f2116e5643c9bf06056c7ca463e41ada523/src/libs/apistore.ts#L22
There's a mod of operation where the style is stored in a "backend" and not it local storage.

Should this be converted to a discussion or is there an issue you have encountered that is problematic?

Also the console log is somewhat descriptive I think:
![image](https://github.com/maplibre/maputnik/assets/3269297/dad7b2a2-2141-495e-882d-ea5071baf752)

**Comment by zstadler** [comment]
There is no functional issue, but it is strange. I didn't expect to see any desktop-related functionality in the web site.
Was there a discussion about on-boarding the desktop?

**Comment by HarelM** [comment]
There was a discussion on desktop to some extent here:
https://github.com/maplibre/maplibre/issues/352
Currently, desktop is created as part of the CI of the web, I'm guessing that their functionality is not "separated" enough.
A build flag is what I would expect in this kind of scenario, it might be that I overlooked it when migrating to vite.

**Comment by zstadler** [comment]
Should a new discussion on Maputnik Desktop be done in this repository, or in the MapLibre organization level?

**Comment by HarelM** [comment]
Depending on what you would like to achieve.
I personally don't think anything else is needed in term of migrating a repository, as I see the desktop a "nice to have" king of a feature.
Having said that, if someone is willing to maintain it, s.he could open an on boarding issue and explain why this part should be in maplibre.
There's also another option to move this part of the code into this repo, but again, it will require someone who understands the code and would like to continue maintaining it.

Maintainence and maintainer are the key word here I guess.

**Comment by nyurik** [comment]
I think desktop should be part of this repo, e.g. `/desktop`, but only if someone can create a pr that:
* moves all the code with history (I can help with that)
* modifies ci to automate build and test
* updates the docs
* updates ci release process to auto publish the binary

**Comment by zstadler** [comment]
I feel that in addition, the Maputnik Desktop also requires some UX changes.

For example when "watching" a local file and then opening a different file from the app, the first file is overwritten with the contents of the second file. Luckily for me, I had the first file also opened in an editor, so I was able to recover.

There is another option for the future of Desktop: 
- Keep desktop as a separate repository
- Move all desktop-specific code to the desktop repository
- Have the maputnik CI/CD build, test and publish https://maplibre.org/maputnik
- Have the desktop CI/CD build, test and publish the desktop app, using code from the maputnik repository

I agree with @nyurik that any future support of desktop requires someone to prepare a PR.

**Comment by hrecht** [comment]
Hello, I found this discussion looking for updates on the desktop app. I love the web app but I've found it difficult to use my own local map tiles and font files in the web version of the app and would use a desktop version if possible.

Is the desktop or local version still supported? If not, it could be helpful to update the documentation. Or if there's a good way to run the latest version of Maputnik locally that would be helpful to document too.

I see the desktop folder was moved last month with https://github.com/maplibre/maputnik/commit/7ac1b03b5a8f240a10578cbb9df06d87d460bb41 from @kevinschaul but the latest app release is from 2020 at https://github.com/maplibre/maputnik/releases/tag/v1.7.0. The wiki still refers to it as an option https://github.com/maplibre/maputnik/wiki/Maputnik-CLI.

If it's helpful, I'd be happy to open a PR with suggested updates to the readme or wiki once there's more info on the desktop app status, either way.
